pyccel/ast/numpyext.py

```diff
@@ -376,21 +375,21 @@ def fprint(self, printer, lhs = None):
 
 def Shape(arg):
     if arg.shape is None:
-        return PythonTuple([PyccelArraySize(arg,i) for i in range(arg.rank)])
+        return PythonTuple(*[PyccelArraySize(arg,i) for i in range(arg.rank)])
```

Comment:
Can we create a PythonTuple from a generator or do we need this *[] ?

Reply:
yes we can create it from a generator

Reply:
Then we should create it from a generator as this is more efficient

Reply:
yes it's more memory efficient